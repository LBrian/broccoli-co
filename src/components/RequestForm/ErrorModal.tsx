import type { ComponentProps } from 'react'
import Modal from '../Modal'

const ErrorModal = ({ onClose, children, ...props }: Omit<ComponentProps<typeof Modal>, 'name'>) => {
  return (
    <Modal name='request-error' {...props} onClose={onClose}>
      <h3 className='font-bold text-lg text-error'>Oops!</h3>
      <p className='py-4 text-error'>{children}</p>
      <div className='modal-action'>
        <button aria-label='Try again' data-testid='try-again' className='btn btn-error' onClick={onClose}>
          Try Again
        </button>
      </div>
    </Modal>
  )
}

export default ErrorModal
