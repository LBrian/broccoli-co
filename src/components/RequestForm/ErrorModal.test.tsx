import { render, screen } from '@testing-library/react'
import userEvent from '@testing-library/user-event'
import ErrorModal from './ErrorModal'

const mockOnClose = jest.fn()

test('renders ErrorModal component', () => {
  render(<ErrorModal isOpen onClose={mockOnClose} />)

  const errorModal = screen.getByTestId('request-error')

  expect(errorModal).toBeInTheDocument()
  expect(errorModal).toBeVisible()
  expect(errorModal).toMatchSnapshot()
})

test('click Try Again button should invoke onClose callback', async () => {
  render(<ErrorModal isOpen onClose={mockOnClose} />)

  expect(mockOnClose).toBeCalledTimes(0)

  const tryAgainBtn = screen.getByTestId('try-again')

  expect(tryAgainBtn).toHaveAttribute('aria-label')
  expect(tryAgainBtn).toHaveTextContent('Try Again')

  await userEvent.click(tryAgainBtn)

  expect(mockOnClose).toBeCalledTimes(1)
})
