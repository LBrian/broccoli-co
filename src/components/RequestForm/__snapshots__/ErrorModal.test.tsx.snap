// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders ErrorModal component 1`] = `
<div
  class="modal-box w-auto py-16 modal-middle"
  data-testid="request-error"
>
  <h3
    class="font-bold text-lg text-error"
  >
    Oops!
  </h3>
  <p
    class="py-4 text-error"
  />
  <div
    class="modal-action"
  >
    <button
      aria-label="Try again"
      class="btn btn-error"
      data-testid="try-again"
    >
      Try Again
    </button>
  </div>
</div>
`;
